qiskit_nature/algorithms/initial_points/mp2_initial_point.py

```diff
@@ -322,7 +313,11 @@ def get_energy_corrections(self) -> np.ndarray:
 
     def get_energy_correction(self) -> float:
         """The overall energy correction."""
-        return self.get_energy_corrections().sum()
+        unique_corrections = {
+            str(np.asarray(exc) % self._integral_matrix.shape[0]): corr
+            for corr, exc in zip(self.get_energy_corrections(), self.excitation_list)
+        }
```

Comment:
I'm not sure I fully where the filtering is going on here.
I assume, the dictionary keys might clash at some point, but do they do some in a controllable manner? I.e. in terms of order? How do you ensure (if they clash) that the correct one is being used?

Reply:
My (probably wrong) assumption was these would always be identical. For example, the excitation with occupied and virtual molecular orbital indicies `((0, 1), (2, 5))` has three corresponding excitations when indexed by spin orbital:
```python
((0, 1), (2, 5)):  9.785199275020633e-08
((0, 1), (2, 5)):  9.785199275020633e-08
((0, 1), (2, 5)):  9.785199275020633e-08
```
My limited computational chemistry knowledge led me to believe this would always be the case, so I just take whichever one is computed last.

Reply:
These are in fact only guaranteed to be identical in the restricted-spin setting (which (right now) is the only setting currently supported by the MP2 initial point anyways)..
Your explicit example did however clear up my question as I failed to recognize before, how the modulo operation was filtering various states.

I'm assuming, that within the assumption of the current restricted-spin formula for the energy correction computation, this works fine. When moving towards unrestricted spin cases, we will need to update both, that formula and this filtering technique however. Maybe leave a comment here in the code, indicating this :+1: